### ISSUE TITLE
Reproduction of Mistral 7B NCA

### ISSUE THREAD

**Issue by kygguo** [thu-ml/Noise-Contrastive-Alignment#4]
Thanks for sharing this wonderful repo! I have quick questions regarding the reproduction of Mistral 7B NCA (for follow-up works).

> We follow [43] and fine-tune all parameters of the pretrained model with UltraInteract and UltraFeedback. Specifically, for HuggingFaceH4/mistral-7B-SFT-beta model, we directly adopt the recipe of Eurus, with the same data mixture (all 220K multi-turn trajectory pairs in UltraInteract and all 340K pairs in UltraFeedback) and hyperparameters.

1. Does this mean that NCA is done directly upon Eurus-7b-sft? 
2. The paper reports training dataset involves UltraInteract and UltraFeedback, but in this repo preference data only contains 62K rows, are they only PART of UltraFeedback? If I am not misunderstanding, I need to manually merge UltraInteract and UltraFeedback together to reproduce. So, one more thing I want to confirm: the UltraFeedback_binaraized dataset only provides (max_score, other_random) as preference pair, but for NCA it seems that all combinational pairs are used, right?

**Comment by ChenDRAG** [comment]
Thank you for the interest in the repo.
1. I'm not entirely sure about this question. Need to check with @lifan-yuan for a precise answer.  
2. No. The training dataset is either UltraInteract (for preference experiments) or UltraFeedback (for reward experiments), not both.   The repo currently only contains training scripts for the UltraFeedback dataset, corresponding to Table 2 Figure 4 in the paper.
3. For reward-based  NCA, we use all combinational pairs, corresponding to the ChenDRAG/ultrafeedback_reward dataset. For preference-based  NCA, we use just a subset of the previous dataset, which is UltraFeedback_binaraized as mentioned.

**Comment by kygguo** [comment]
Thanks for the quick response!

RE 2: How about the results in Table 3? According to the paper, it seems to use both UltraInteract and UltraFeedback as training data. Do you have any plan to release more implement details regarding this experiment?

**Comment by ChenDRAG** [comment]
Results of Table 3 are all based on UltraInteract, which is not contained in this repo. 
Currently I have no plan to support UltraInteract training. The base model and dataset are all publicly accessible, though.   The core algorithm is exactly the same as is implemented in this repo.

I would recommend keeping track of https://github.com/OpenBMB/Eurus for possible future support of UltraInteract training scripts.

**Comment by kygguo** [comment]
Thanks for all the clarification!

**Comment by kygguo** [comment]
Thanks for all the clarification!
